src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3612,21 +3612,27 @@ void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
                         // the value has been evaluated. So we assemble the reference after visiting the value.
 
                         _evalStack.Push(Visit(memberReference.Instance));
-                        return (success: true, propertyArguments);
-
+                        return (success: true, propertyArguments.CastArray<IOperation>());
 
                     case OperationKind.ArrayElementReference:
-                    // PROTOTYPE(dataflow): Add support
+                        var arrayReference = (IArrayElementReferenceOperation)instance;
+                        ImmutableArray<IOperation> indicies = arrayReference.Indices.SelectAsArray(index =>
```

Comment:
Consider indexExpr over index. Given the use of int as ID here would be good to avoid names that are commonly associated with int types.